python/docs/lis/userguide.rst

```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
+mnemonics and the units of measurement by simply looping the Spec Blocks:
+
+.. code-block:: python
+
+    >>> for spec in format_spec.specs:
+    ...     print("Index: {} (units={})".format(spec.mnem, spec.units))
+    "Index: 'TDEP', units: '.1IN')"
+    "Index: 'TIME', units: 'ms  ')"
+    "Index: 'CHX ', units: 'M   ')"
+    "Index: 'RCCL', units: '    ')"
 
 .. note::
-   LIS79 opens for the presence of duplicated DFSR, for redundancy. Currently,
-   dlisio has no support for identifying redundant DFSRs, and curves() will
-   return empty arrays for these.
+    Please refer to :class:`dlisio.lis.DataFormatSpec` and
+    :class:`dlisio.core.spec_block_0` / :class:`dlisio.core.spec_block_1` for
+    the complete documentation of DFSR and Spec Blocks, respectively.
 
+Reading the curves
+..................
+
+The :class:`dlisio.lis.DataFormatSpec` only contains the metadata about the
+logset. To read the actual curves, pass the DFSR ``format_spec`` to
+:func:`dlisio.lis.curves` together with the filehandle ``f``:
+
+.. code-block:: python
+
+    >>> curves = lis.curves(f, format_spec)
+
+This returns a `structured numpy.ndarray`_ containing all the curves in the
+logset, using the mnemonics from the Spec Blocks as field (column) names.
+
+Fast Channels
+.............
+
+There is one caveat to reading the curves from a logset that is necessary to be
+aware of. LIS supports a concept it refers to as 'Fast Channels'. In short this
+means that channels within the *same* logset can have different sampling rates.
+
+When a logset contains curves with different sampling rates, these have to be
+read separately. I.e. multiple calls to the :func:`dlisio.lis.curves` is
+necessary in order to read all the curves. The sampling rate of a channel is
+recorded in its Spec Block:
+
+.. code-block:: python
+
+    >>> ch = format_spec.specs[-1]
+    >>> ch.mnem
+    'RCCL'
+    >>> ch.samples
+    6
+
+The last channel in this Data Format Spec, ``RCCL``, has a sampling rate of 6,
+which means it's sampled 6 times as frequent as the index channel. Remember,
+that the index ``DEPT`` was sampled once every 6th inch. That implicitly means
+that ``RCCL`` is sampled once every inch.
+
+You can also get a set of all sample rates in the DataFormatSpec:
+
+.. code-block:: python
+
+    >>> format_spec.sample_rates()
+    {1, 6, 30}
+
+There are 3 different sampling rates, which be must read with separate calls to
+:func:`dlisio.lis.curves()`:
+
+.. code-block:: python
+
+    >>> data_01 = lis.curves(f, format_spec, sample_rate=1)
+    >>> data_06 = lis.curves(f, format_spec, sample_rate=6)
+    >>> data_30 = lis.curves(f, format_spec, sample_rate=30)
+
+.. note::
+    Please refer to :func:`dlisio.lis.curves` for a more comprehensive
+    explanation of this topic.
+
+.. note::
+    When reading channels with a higher sampling rate, the index is still
+    included in the resulting numpy array, and the missing values have been
```

Comment:
the missing vales (are) linearly interpolated.

(time consistency with "index is included".)